packages/google_sign_in/google_sign_in/lib/google_sign_in.dart

```diff
@@ -424,4 +432,10 @@ class GoogleSignIn {
     await _ensureInitialized();
     return GoogleSignInPlatform.instance.requestScopes(scopes);
   }
+
+  /// Checks if the given `accessToken` authorized to access all `scopes`.
+  Future<bool> canAccessScopes(String? accessToken, List<String> scopes) async {
+    await _ensureInitialized();
+    return GoogleSignInPlatform.instance.canAccessScopes(accessToken, scopes);
```

Comment:
I *think* so, here's the Android version:

* https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignIn#hasPermissions(com.google.android.gms.auth.api.signin.GoogleSignInAccount,%20com.google.android.gms.common.api.Scope...)

And on iOS:

* https://developers.google.com/identity/sign-in/ios/reference/Classes/GIDGoogleUser#grantedscopes

As you suggested above, the token may be a web-only artifact, so I'm going to make it a completely optional, named parameter so it can be ignored by other implementations if you ever need to implement this other than "isSignedIn".